tools/autograd/gen_variable_type.py

```diff
@@ -948,6 +985,17 @@ def emit_body(
     base_name = get_base_name(f)
     view_info = get_view_info(f)
 
+    is_inplace_foreach = name.startswith("_foreach") and inplace
+
+    inplace_foreacharg2refarg: Dict[Argument, Argument] = {}
+    if is_inplace_foreach and info is not None:
+        for foreach_arg, ref_arg in zip(
+            f.func.arguments.flat_non_out, info.func.func.arguments.flat_non_out
+        ):
+            foreach_arg_type = getattr(foreach_arg.type, "elem", foreach_arg.type)
+            assert foreach_arg_type == ref_arg.type
+            inplace_foreacharg2refarg[foreach_arg] = ref_arg
+
```

Comment:
I think its better to replace`inplace_foreacharg2refarg` with `is_inplace_foreach` and then doing `assert info is not None` inside instead (if possible, if there some problematic ops I'm curious what those are). The info is not None case should be handled elsewhere.

Reply:
hmm, I thought it was enough because a container of 0 elements (here `inplace_foreach2refarg` is a dictionary) is evaluated as `False` in Python. I do know I'm biased here as I've played with this thing. I'll follow your suggestion